app/models/concerns/contents_list.rb

```diff
@@ -0,0 +1,104 @@
+module ContentsList
+  extend ActiveSupport::Concern
+
+  MINIMUM_CHARACTER_COUNT = 415
+  MINIMUM_CHARACTER_COUNT_IF_IMAGE_PRESENT = 224
+  MINIMUM_TABLE_ROW_COUNT = 13
+  MINIMUM_TABLE_ROW_COUNT_IF_IMAGE_PRESENT = 6
+
+  included do
+    def contents
+      @contents ||=
+        if show_contents_list?
+          contents_items.each { |item| item[:href] = "##{item[:id]}" }
+        else
+          []
+        end
+    end
+
+    def contents_items
+      extract_headings_with_ids
```

Comment:
This line could do with a test - unless it's never expected to be called (always overwritten by the class the module is included in) - in which case we could maybe remove it entirely?

Reply:
I had a look at the test coverage and [the method itself](https://github.com/alphagov/frontend/pull/4608/commits/ef5ef221c521c6b84d0bf8ec4e809aab5a898ea3#diff-73e131976ad489a72d889fad2320885cced2b1c302b0c89373ea30d874f3cd4eR44-R50) is covered but not this specific line as `contents_items` is always overwritten for this document.

Some of the other document types don't overwrite this method, so this line should hopefully be covered by their tests when they are moved across.